### PR TITLE
Integrate checked `StrictMVars` from `strict-checked-vars`

### PR DESCRIPTION
**Pull request by jorisdral**
# Description

Add `StrictMVar`s with default `NoThunks` invariants. Invariants, including `NoThunks` invariants, can be switched on/off at build time using the `checkmvarinvariants` cabal flag for the `strict-checked-vars` package.

Replace `StrictSVar`s by `StrictMVar`s where possible.

### CODE REVIEW

**Review by amesgen — APPROVED** [review]
Great to see unnecessary `StrictSVar` usage being removed :tada: 

AFAICT, there are only two remaining usage sites of `StrictSVar`:

 1. In the ImmutableDB:
    ```haskell
    varInternalState :: !(StrictSVar m (InternalState m blk h))
    ``` 
    The implementation currently relies on the special `readSVar` behavior (returning the most recent value, even if the `SVar` is currently empty), so we want to keep using `StrictSVar` here.
 2. In `Ouroboros.Consensus.Protocol.Ledger.HotKey.evolveKey`.

    I think whether to use `StrictSVar` or `StrictMVar` depends on whether other KES operations should block or use the most recent key while the key is evolving; I haven't yet thought deeply about which semantics we want (I don't even have an intuition how much wallclock time key evolution takes, I imagine it is quite small?).

File: test.log
[the remark this answers is not in the document]
Comment (jorisdral) [reply test.log:1]: Oops, removed!

File: ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/VolatileDB/StateMachine.hs [hunk 1]
```diff
@@ -479,7 +480,7 @@ shrinkCmd _ cmd = case cmd of
 -- | Environment to run commands against the real VolatileDB implementation.
 data VolatileDBEnv = VolatileDBEnv
   { varErrors :: StrictTVar IO Errors
-  , varDB     :: StrictSVar IO (VolatileDB IO Block)
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/VolatileDB/StateMachine.hs:482]: Good point. I would also be surprised if it has an effect on performance, so I'll replace this by a `StrictTVar`!

File: ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ImmutableDB/StateMachine.hs [hunk 2]
```diff
@@ -800,15 +801,15 @@ data ImmutableDBEnv = ImmutableDBEnv {
       -- During truncation we might need to delete a file that is still opened
       -- by an iterator. As this is not allowed by the MockFS implementation, we
       -- first close all open iterators in these cases.
-    , varDB     :: StrictSVar IO ImmutableDBState
+    , varDB     :: StrictMVar IO ImmutableDBState
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ImmutableDB/StateMachine.hs:804]: See https://github.com/input-output-hk/ouroboros-consensus/pull/332#discussion_r1315153813

File: ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/StateMachine.hs [hunk 3]
```diff
@@ -352,7 +353,7 @@ data ChainDBState m blk = ChainDBState
 
 -- | Environment to run commands against the real ChainDB implementation.
 data ChainDBEnv m blk = ChainDBEnv {
-    varDB           :: StrictSVar m (ChainDBState m blk)
+    varDB           :: StrictMVar m (ChainDBState m blk)
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/StateMachine.hs:356]: See https://github.com/input-output-hk/ouroboros-consensus/pull/332#discussion_r1315153813

**Review by amesgen — APPROVED** [review]
LGTM

**Pull request merged**